apps/platforms/linux/terminator/terminator-linux.py

```diff
@@ -0,0 +1,82 @@
+from talon import Context, actions, Module
+
+# App definition
+mod = Module()
+mod.apps.terminator = """
+os: linux
+and app.exe: terminator
+os: linux
+and app.name: Terminator
+"""
+
+# Context matching
+ctx = Context()
+ctx.matches = r"""
+app: terminator
+"""
+ctx.tags = [
+    'terminal',
+    'user.terminal',
+    'user.tabs',
+    'user.splits',
+    'user.generic_unix_shell',
+    'user.git',
+    'user.kubectl',
+]
+
+
+# --- Implement actions ---
+@ctx.action_class('user')
+class user_actions:
+    # user.splits
+    def split_window_right(): actions.key('alt-right')
+    def split_window_left(): actions.key('alt-left')
+    def split_window_down(): actions.key('alt-down')
+    def split_window_up(): actions.key('alt-up')
+    def split_window_vertically(): actions.key('shift-ctrl-e')
+    def split_window_horizontally(): actions.key('shift-ctrl-o')
+    def split_flip(): actions.key('super-r')
+    def split_window(): actions.key('shift-ctrl-o')
+    def split_clear(): actions.key('shift-ctrl-r')
+    def split_clear_all(): actions.key('shift-ctrl-g')
+    def split_next(): actions.key('shift-ctrl-n')
+    def split_last(): actions.key('shift-ctrl-p')
+
+
+@ctx.action_class('app')
+class AppActions:
+
+    # app.tabs
+    def tab_open(): actions.key('ctrl-shift-t')
+    def tab_previous(): actions.key('ctrl-pageup')
+    def tab_next(): actions.key('ctrl-pagedown')
+    def tab_close(): actions.key('ctrl-shift-w')
+    # global (overwrite linux/app.py)
+    def window_open(): actions.key('ctrl-shift-i')
+    def window_close(): actions.key('ctrl-shift-q')
+
+
+# global (overwrite linux/edit.py)
+@ctx.action_class('edit')
+class EditActions:
+    def page_down(): actions.key('shift-pagedown')
+    def page_up(): actions.key('shift-pageup')
+    def paste(): actions.key('ctrl-shift-v')
+    def copy(): actions.key('ctrl-shift-c')
+
+    def find(text: str = None):
+        actions.key('ctrl-shift-f')
+        if text:
+            actions.insert(text)
+
+    def delete_line():
+        actions.edit.line_start()
+        actions.key('ctrl-a ctrl-k')
```

Comment:
This seems like something that should be in `generic_unix_shell` instead, especially since this isn't the terminal emulator's behaviour but whatever program is running within.

Reply:
Is that so? This deviates from the key binding used in the gnome terminal, at least. So I was concerned it was terminator specific?

Reply:
Is the ctrl-a necessary in terminator? If so, then this does look to be app specific.

Reply:
`ctrl-a` would just go to the beginning of the line in readline/emacs, right?

I'll note too that this is definitely wrong in a shell with vim keybindings.

Reply:
Here is the documentation that specifies the shortcuts, which I used as reference. If I try to use control k without first having pressed control a I get no response. https://terminator-gtk3.readthedocs.io/en/latest/gettingstarted.html

Reply:
That page doesn't list `ctrl-a` nor `ctrl-k`. Those are both standard readline keybindings. `ctrl-k` cuts from the cursor position to the end of the line buffer.

As an aside, `ctrl-u` cuts from the cursor to the beginning of the line buffer. Again, this is standard readline behaviour. Terminal emulators do not have a concept of a line input buffer.

Reply:
Yeah, on review I see that it's not one of the shortcuts I pulled by looking into the terminated documentation. It's a method I implemented because it's inside the file for the Linux gnome terminal. Considering the keybinding is different in the gnome implementation, I was sure that this should be moved to a generic editor? If so, I'll move it there.

Reply:
Moving this to a tag context probably warrants further discussion in a separate PR. Not everyone uses emacs keybindings in their shell.

Reply:
Is it the case then, that I should not implement this method all together?